### DISCUSSION TITLE
[Feature] More Training Mode Options

### DISCUSSION THREAD

**Discussion by TruXail** [ikemen-engine/Ikemen-GO#888] [Feature Request] [answered]
Within the training menu within the pause menu, it would be cool and quite helpful to have additional options. These options include but are not limited to: 

1. Input copy mode. where the player is to set an input within a time frame and the training dummy will repeat the input until stated otherwise.

2. Response mode, where the training dummy is to respond to an attack with a certain button (e.g when hit with an attack, respond with a or x as soon as possible, but only when hit.

3. Specific distances, an option that sets the training dummy standing distance to the current standing distance as close puts the training dummy extremely close to the player, and medium can at times be too far to gain significant worth.

4. Specific placement (aka, return mode). an option where the training dummy will move/return to a set position once a certain time is elapsed after not being hit, or after a certain time frame after preforming an input from the copy input response mode.

5. Instant reload, both the training dummy and the player return to a neutral state as if beginning a new round after pressing a certain button/input to initiate the instant reload

**Comment by Orochikyocr** [comment]
All this stuff can already be done by modifying the training.zss file with a basic knowledge of ZSS.

The only thing we really need is K4thos to finish setting up the pause menu and document how can people easily add extra buttons to the pause menu. Maybe an LUA file inside the mod folder ready to add custom buttons that can change map and vars values. 
I already did add extra menu items it was a pain in the ass, without K4thos help I won't be able to do this. 

But positioning, counter, and 1st hit guard are way easier. This stuff can be shared between users using a custom ZSS file, no need for overwriting core files.
TBH adding too much stuff to the training mode is just more BUGS to fix, let that job to high-level users.

https://streamable.com/3klrak

This video shows all those stuff already working.

**Comment by TruXail** [comment]
Whilst this is on the right track of what I'm talking about its not even close to what I'm envisioning. The only thing this video, https://streamable.com/3klrak , has which is similar to what I'm talking about is:

5. Instant reload, both the training dummy and the player return to a neutral state as if beginning a new round after pressing a certain button/input to initiate the instant reload

Every other feature from this list is missing, so unfortunately this is not what i am looking for.

**Comment by potsmugen** [comment, marked as answer]
While you're probably requesting too many features, which could make it more complicated than it needs to be, I was about to suggest that the guard modes could be expanded to "None", "Auto" or "All". Where "All" is how "Auto" currently works, and the new "Auto" being briefly activating autoguard if the player drops a combo.

Fairly common feature in fighting games.

I think the input record/playback would be neat too, but that probably comes after/with the new input parser.
